src/operators/append.ml

```diff
@@ -125,6 +125,11 @@ object (self)
           source#seek n
       | `Append s -> s#seek n 
 
+  method self_sync =
```

Comment:
You should add a comment for future generations stating that this is an approximation...

Reply:
I'll add it ad the `source.mli` level.